### DISCUSSION TITLE
DIPimage version

### DISCUSSION THREAD

**Discussion by AndreZeug** [DIPlib/diplib#143] [General] [answered]
Is there an easy way to find the version of DIPimage / DIPlib used in MATLAB?
Since I move from 2.9 to 3.4.1, I need to update a lot of functions. To make it compatible to both versions I would like to find an easy way to identify the version of DIPimage of the running instance to use something like
```
if myDipVer >=3
  int99 = percentile(dip_image(B), 99));
  ...
else
  int99 = double(dip_percentile(B, [], 99, []));
  ...
end
```
but I only found (not recommended)
`myDipVer = ver('DIPimage');  myDipVer = str2double(myDipVer.Version(1:3))`
to get the version of DIPimage.
Is there a recommended way?

**Comment by crisluengo** [comment, marked as answer]
Using `ver('DIPimage')` is a good approach. The only problem with `str2double(myDipVer.Version(1:3))` is that this will stop working for version 3.10. I would consider looking only at the first character, see if it's a 2 or a 3.

`strsplit(ver('DIPimage').Version,'.')` returns a cell array `{'major', 'minor', 'patch'}`.

`cellfun(@str2double,strsplit(ver('DIPimage').Version,'.'))` returns a regular numeric array `[major, minor, patch]`.

[Note that the syntax `ver('DIPimage').Version` works only in fairly recent MATLAB versions, for older versions you need to assign the output of `ver()` to a variable before you can do the dot indexing.]

You could also do something like `exist('dip_percentile','file')` when choosing to call `dip_percentile` vs `percentile` (though `percentile` also worked in 2.9), or `exist('dip_initialize','file')` for a more generic "is this DIPimage v2 or v3?".
